presto-hive/src/test/java/io/prestosql/plugin/hive/TestHiveIntegrationSmokeTest.java

```diff
@@ -7549,6 +7549,80 @@ public void testSelectFromViewWithoutDefaultCatalogAndSchema()
         assertQuery(sessionNoCatalog, "SELECT count(*) FROM hive.tpch." + viewName, "VALUES 1");
     }
 
+    @Test(invocationCount = 100)
+    public void testSelectFromPrestoViewReferencingHiveTableWithTimestamps() {
+        // Hive views tests covered in TestHiveViews.testTimestampHiveView and TestHiveViesLegacy.testTimestampHiveView
+        String hiveTableName = "select_from_presto_view_hive_table_timestamps_table_" + randomTableSuffix();
+        assertUpdate(
+                withTimestampPrecision(getSession(), HiveTimestampPrecision.NANOSECONDS),
+                "CREATE TABLE " + hiveTableName + " AS SELECT TIMESTAMP '1990-01-02 12:13:14.123456789' ts",
+                1);
+
+        Session defaultSession = getSession();
+        Session millisSession = Session.builder(getSession())
+                .setCatalogSessionProperty("hive", "timestamp_precision", "MILLISECONDS")
+                .setCatalogSessionProperty("hive_timestamp_nanos", "timestamp_precision", "MILLISECONDS")
+                .build();
+        Session nanosSessions = Session.builder(getSession())
+                .setCatalogSessionProperty("hive", "timestamp_precision", "NANOSECONDS")
+                .setCatalogSessionProperty("hive_timestamp_nanos", "timestamp_precision", "NANOSECONDS")
+                .build();
+
+        // Presto view created with config property set to MILLIS and session property not set
+        String prestoViewNameDefault = "select_from_presto_view_hive_table_timestamps_default_view_" + randomTableSuffix();
+        assertUpdate(defaultSession, "CREATE VIEW " + prestoViewNameDefault + " AS SELECT *  FROM " + hiveTableName);
+
+        assertQuery(defaultSession, "SELECT CAST(ts as VARCHAR) FROM " + prestoViewNameDefault, "VALUES ('1990-01-02 12:13:14.123')");
```

Comment:
if you use `assertThat(query(...))`  you become free of H2 and then you can use `TIMESTAMP` literals in the expected query. This would avoid need for `CAST AS varchar)` altogether.

(here we don't have JDBC, and we have more trust implicit trust in our internal APIs; all engine timestamp tests depend on them anyway)

Reply:
Like that?
```
assertThat(query(defaultSession, "SELECT ts FROM " + prestoViewNameDefault)).matches("VALUES TIMESTAMP '1990-01-02 12:13:14.123'");
```